src/structured-data/structured-data-tags.js

```diff
@@ -506,6 +506,9 @@ export function StructuredDataTags({ structuredData = {} }, config, pageType, re
   if (!isStructuredDataEmpty && pageType === "story-page-amp") {
     const newsArticleTags = generateNewsArticleTags();
     newsArticleTags ? tags.push(storyTags(), newsArticleTags) : tags.push(storyTags());
+    if (story["story-template"] === "visual-story") {
```

Comment:
It would be beneficial to perform sanity checks on other story templates alongside testing the visual story schema.